### PR TITLE
Babel

### PR DESCRIPTION
**Pull request by josepplloo**
## Summary

 - 3 / 10 test solved
## Screenshots

image

## Trello Card

https://trello.com/c/1QZG63Um/6-js-stack-ii-ecma-6-7

### CODE REVIEW

File: react/finger-exercises/ecmascript/01 - spread/index.js [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+  }
+  return Math.min(...args);
 }
 
-export function copy() {
+export function copy(args) {
+  if (isArray(args)) {
+    return [...args];
+  }
+  return Object.assign({}, args);
```
Comment (cvega08) [inline react/finger-exercises/ecmascript/01 - spread/index.js]: Here you can use spread operator to copy args object {...args} and use a ternary operator for it 

``` return isArray(args) ? [...args] : { ...args }; ```

File: react/finger-exercises/ecmascript/01 - spread/index.js [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
-  isArray();
-  return Math.min();
+export function min(...args) {
+  if (args[0] === undefined) {
+    return undefined;
+  }
+  if (isArray(args[0])) {
+    return Math.min(...args[0]);
+  }
+  return Math.min(...args);
```
Comment (erikankodelia) [inline react/finger-exercises/ecmascript/01 - spread/index.js]: You can change it to something like this: 

const values = isArray(args[0]) ? args[0] : args;
return Math.min(...values)

File: react/finger-exercises/ecmascript/01 - spread/index.js [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+export function copy(args) {
+  if (isArray(args)) {
+    return [...args];
+  }
+  return Object.assign({}, args);
+}
 
+export function reverseMerge(arrOne, ArrTwo) {
+  const reverseMergeArr = [...ArrTwo, ...arrOne];
+  return reverseMergeArr;
```
Comment (erikankodelia) [inline react/finger-exercises/ecmascript/01 - spread/index.js]: You can return directly [...ArrTwo, ...arrOne]

File: react/finger-exercises/ecmascript/01 - spread/index.js [hunk 4]
```diff
@@ -1,10 +1,36 @@
...
+  }
+  return Object.assign({}, args);
+}
+
+export function reverseMerge(arrOne, ArrTwo) {
+  const reverseMergeArr = [...ArrTwo, ...arrOne];
+  return reverseMergeArr;
+}
+
+export function filterAttribs(args) {
```
Comment (cvega08) [inline react/finger-exercises/ecmascript/01 - spread/index.js]: All this is unnecessary I recommend use destructuring to filter always a and b.

Comment (josepplloo) [reply react/finger-exercises/ecmascript/01 - spread/index.js]: You Right! 
```
export function filterAttribs(args) {
  const { a, b, ...filterObj } = args;
  return filterObj;
}
```

Comment (cvega08) [reply react/finger-exercises/ecmascript/01 - spread/index.js]: Now you can apply destructuring on args like export function filterAttribs({a,b, ...filterObj}) {...}

File: react/finger-exercises/ecmascript/02 - promises/index.js [hunk 5]
```diff
@@ -1,9 +1,34 @@
...
+        const patient = Error('This time is too much !');
+        reject(patient);
+      }
+    }, 100);
+  });
 }
 
-export function asyncDelay() {
-
+export function asyncDelay(period) {
```
Comment (cvega08) [inline react/finger-exercises/ecmascript/02 - promises/index.js]: Here you can call delay function passing period argument. So you don't repeat code

File: react/finger-exercises/ecmascript/02 - promises/index.js [hunk 6]
```diff
@@ -1,9 +1,34 @@
...
+  return new Promise((resolve, reject) => {
+    setInterval(() => {
+      const endTime = Date.now();
+      if (endTime - startTime > period) {
+        resolve(endTime - startTime);
+      } else if (endTime - startTime > MAX_TIME) {
+        const patient = Error('This time is too much !');
+        reject(patient);
+      }
+    }, 100);
```
Comment (cvega08) [inline react/finger-exercises/ecmascript/02 - promises/index.js]: I think you can use 'period' for this.

File: react/finger-exercises/ecmascript/02 - promises/index.js [hunk 7]
```diff
@@ -1,9 +1,34 @@
 // Hint: use setInterval, create a new Promise and measure time with Date.now()
 
-export function delay() {
-
+export function delay(period) {
+  const startTime = Date.now();
+  const MAX_TIME = 4900;
+  return new Promise((resolve, reject) => {
```
Comment (cvega08) [inline react/finger-exercises/ecmascript/02 - promises/index.js]: Here you always execute setInterval but if period is bigger than maxTime you can reject without setInterval 

```const startTime = Date.now();
 const maxTime = 4900;
 return new Promise((resolve, reject) => {
    if (periodo < maxTime)
      setInterval(...)
    else 
      ....
  })``

File: react/finger-exercises/ecmascript/01 - spread/index.js [hunk 8]
```diff
@@ -1,10 +1,21 @@
 import { isArray } from './utils';
 
-export function min() {
-  isArray();
-  return Math.min();
+export function min(...args) {
+  if (args.length === 0) {
```
Comment (cvega08) [inline react/finger-exercises/ecmascript/01 - spread/index.js]: Delete === 0 with this (!args.length) it should be work

File: react/finger-exercises/ecmascript/02 - promises/index.js [hunk 9]
```diff
@@ -1,9 +1,23 @@
...
+        resolve(endTime - startTime);
+      }
+    }, period);
+  });
 }
 
-export function asyncDelay() {
-
+export async function asyncDelay(period) {
+  const result = await delay(period);
```
Comment (cvega08) [inline react/finger-exercises/ecmascript/02 - promises/index.js]: return await delay(period);

**Pull request merged**